src/rad/appdata.cpp

```diff
@@ -1319,16 +1321,41 @@ bool ApplicationData::LoadProject( const wxString &file, bool justGenerate )
 
 		if ( older )
 		{
+			wxLogDebug("Old version: %d.%d -- current %d.%d", fbpVerMajor, fbpVerMajor, m_fbpVerMajor, m_fbpVerMinor);
+
 			if( justGenerate ){
 				wxLogError( wxT( "This project file is out of date.  Update your .fbp before using --generate" ) );
 				return false;
 			}
-			if ( wxYES == wxMessageBox( wxT( "This project file is not of the current version.\n" )
-			                            wxT( "Would you to attempt automatic conversion?\n\n" )
-			                            wxT( "NOTE: This will modify your project file on disk!" ), _( "Old Version" ), wxYES_NO ) )
+
+			wxFileName fname(file);
+			wxString   strMsg, strOldBackupName(fname.GetFullName() + wxT(".bak"));
+
+			if (wxMessageBox(strMsg.Format(_("This project file is an old version.\n\n"
+			                                 "Do you want to copy the old version to %s and save an updated version?"),
+			                               strOldBackupName),
+			                 _("Old project file"), wxYES_NO) == wxYES)
 			{
-				// we make a backup of the project
-				::wxCopyFile( file, file + wxT( ".bak" ) );
+				{
+					wxLogNull logNo;  // Prevents wxCopyFile from displaying a wxWidgets error.
+					fname = file + wxT(".bak");
+
+					while (!::wxCopyFile(file, fname.GetFullPath()))
+					{
+						wxRichMessageDialog dlg(
+						    NULL, strMsg.Format(_("Unable to save old version to %s.\n\n"), strOldBackupName),
+						    _("Backup failed"), wxICON_WARNING | wxYES_NO | wxCANCEL);
+						dlg.SetYesNoCancelLabels(_("Retry"), _("Skip"), _("Cancel"));
+						dlg.ShowDetailedText(wxSysErrorMsg());
+						auto result = dlg.ShowModal();
+						if (result == wxID_NO)
+							break;  // User wants to skip backing up the file.
+						else if (result == wxID_CANCEL)
+							return false;  // User is cancelling converting the project.
+						else
+							continue;  // wxYES means retry.
```

Comment:
Please use braces also for one liners. I'm not a big fan of these inline comments, i would prefer them above the line they refer to. These are also no complete sentences so please to trailing point.